app_legacy/Helpers/database/player-history.php

```diff
@@ -52,49 +52,39 @@ function getUserBestDaysList($user, $listOffset, $maxDays, $sortBy): array
     return $retVal;
 }
 
-function getAchievementsEarnedBetween($dateStart, $dateEnd, $user): array
+function getAchievementsEarnedBetween($dateStart, $dateEnd, $username): array
 {
-    $retVal = [];
-
-    if (!isValidUsername($user)) {
-        return $retVal;
+    if (!isValidUsername($username)) {
+        return [];
     }
 
-    sanitize_sql_inputs($dateStart, $dateEnd, $user);
-
-    // $dateStrStart = date( "Y-m-d H:i:s", strtotime( $dateStart ) );
-    // $dateStrEnd = date( "Y-m-d H:i:s", strtotime( $dateEnd ) );
-    $dateStrStart = $dateStart;
-    $dateStrEnd = $dateEnd;
+    $bindings = [
+        'dateStart' => $dateStart,
+        'dateEnd' => $dateEnd,
+        'username' => $username,
+        'achievementType' => AchievementType::OfficialCore,
+    ];
 
     $query = "SELECT aw.Date, aw.HardcoreMode, ach.ID AS AchievementID, ach.Title, ach.Description, ach.BadgeName, ach.Points, ach.Author, gd.Title AS GameTitle, gd.ImageIcon AS GameIcon, ach.GameID, c.Name AS ConsoleName
               FROM Awarded AS aw
               LEFT JOIN Achievements AS ach ON ach.ID = aw.AchievementID
               LEFT JOIN GameData AS gd ON gd.ID = ach.GameID
               LEFT JOIN Console AS c ON c.ID = gd.ConsoleID
-              WHERE User = '$user' AND ach.Flags = " . AchievementType::OfficialCore . "
-              AND Date BETWEEN '$dateStrStart' AND '$dateStrEnd'
+              WHERE User = :username AND ach.Flags = :achievementType
+              AND Date BETWEEN :dateStart AND :dateEnd
               ORDER BY aw.Date
               LIMIT 500";
 
-    $dbResult = s_mysql_query($query);
+    $cumulativeScore = 0;
 
-    if ($dbResult !== false) {
-        $achCount = 0;
-        $cumulScore = 0;
-        while ($db_entry = mysqli_fetch_assoc($dbResult)) {
-            $cumulScore += $db_entry['Points'];
+    return legacyDbFetchAll($query, $bindings)
+        ->map(function ($entry) use (&$cumulativeScore) {
+            $cumulativeScore += (int) $entry['Points'];
+            $entry['CumulScore'] = $cumulativeScore;
 
-            $retVal[$achCount] = $db_entry;
-            $retVal[$achCount]['CumulScore'] = $cumulScore;
-
-            $achCount++;
-        }
-    } else {
-        log_sql_fail();
-    }
-
-    return $retVal;
+            return $entry;
+        })
+        ->toArray();
```

Comment:
So your plan to "test and refactor to Laravel models" is just replacing `s_mysql_query` with `legacyDbFetchAll`?
I was expecting to see something like:
```
$user->playerAchievements()->with('GameData')->with('Console')
    ->published()->whereBetween('Date', $dateStart, $dateEnd)
    ->orderBy('Date')->limit(500)->map(...)
```
(pardon my psuedo-code if that's not 100% right).

Though I agree that getting the tests in place is the more important change.

Reply:
Yes, initially. I rewrote some of those queries to fluent eloquent queries but it would take me even longer to finish this task.

Additionally, when moving to proper eloquent queries I'd also want to refactor return types from arrays to collections which, again, would make this an even longer process. One step after another.

Reply:
> Though I agree that getting the tests in place is the more important change.

That's the goal for now, yes. Note that this whole legacyDb* functions approach is just an intermediate step. Eventually most of it should be fluent eloquent, moved to controllers and views, encapsulated into dynamic components, etcetc